### PR TITLE
Add some more logic to smart banner

### PR DESCRIPTION
**Pull request by nramadas**

### CODE REVIEW

File: src/lib/smartBannerState.es6.js [hunk 1]
```diff
@@ -1,21 +1,114 @@
...
 
+const PAGE_PERCENTAGES = {
+  'comments.subreddit': 5,
+};
+
+const USE_TUNE = 100;
+
+const checkDeviceType = (allowedAgents, userAgentString) => {
+  return !!allowedAgents
+    .map(a => userAgentString.indexOf(a) > -1)
```
Comment (schwers) [inline src/lib/smartBannerState.es6.js]: You can use `Array.some` instead of map -> filter -> length check

Comment (nramadas) [inline src/lib/smartBannerState.es6.js]: good point

File: src/lib/smartBannerState.es6.js [hunk 2]
```diff
@@ -1,21 +1,114 @@
...
+
+  // 4) Check the user agent
+  if (!checkDeviceType(ALLOWED_DEVICES, userAgent)) { return BASE_VAL; }
+
+  // 5) only show the banner to people from certain countries
+  if (!ALLOWED_COUNTRIES.has(country)) { return BASE_VAL; }
+
+  // Create a bucket; a few rules are going to depend on that
+  let userId = '';
+  if (loid || data.user) { userId = loid || data.user.id; }
```
Comment (schwers) [inline src/lib/smartBannerState.es6.js]: :no_bell: When do we have neither?

Comment (nramadas) [inline src/lib/smartBannerState.es6.js]: i don't know, and I don't think we ever will. But, i don't want to take any chances either

File: src/lib/smartBannerState.es6.js [hunk 3]
```diff
@@ -1,21 +1,114 @@
...
+  const userIdSum = userId.split('').reduce((sum, chr) => sum + chr.charCodeAt(0), 0);
+  const bucket = userIdSum & 100;
```
Comment (schwers) [inline src/lib/smartBannerState.es6.js]: Should this be `userIdSum % 100`?

Comment (nramadas) [inline src/lib/smartBannerState.es6.js]: i'm an idiot

Comment (schwers) [inline src/lib/smartBannerState.es6.js]: False

**Comment by schwers** [comment]
inb4 we need to add page source to TUNE urls

File: assets/js/client.es6.js [hunk 4]
```diff
@@ -99,28 +90,19 @@ function initialize(bindLinks) {
...
-
-    if (featureEnabled && downloadUrl && inBucket && inCountry) {
-      const { IMPRESSION, CLICK } = downloadUrl;
-      initBanner(IMPRESSION, CLICK);
-    }
+    const {
+      show,
+      impressionUrl,
+      clickUrl,
+    } = shouldShowBanner({
```
Comment (schwers) [inline assets/js/client.es6.js]: :no_bell: Super-Nit:tm: `shouldShowBanner` as name _feels_ like it should be returning a bool versus an object that describes the banner-showing-state. But I don't have any replacement suggestions atm

**Comment by schwers** [comment]
:nail_care: Just a small math nit

**Comment by nramadas** [comment]
:haircut:

**Comment by schwers** [comment]
:fish:

**Pull request merged**